src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -157,52 +150,18 @@ class SourcesTree extends Component<Props, State> {
       return this.setState({ highlightItems });
     }
 
-    if (nextProps.sources === this.props.sources) {
-      return;
-    }
-
-    if (nextProps.sources.size === 0) {
-      // remove all sources
-      this.setState(
-        createTree(
-          nextProps.sources,
-          nextProps.debuggeeUrl,
-          nextProps.projectRoot
-        )
-      );
+    if (nextProps.sources == this.props.sources) {
       return;
     }
 
     // TODO: do not run this every time a source is clicked,
     // only when a new source is added
-    const next = Set(nextProps.sources.valueSeq());
-    const prev = Set(this.props.sources.valueSeq());
-    const newSet = next.subtract(prev);
-
-    const uncollapsedTree = this.state.uncollapsedTree;
 
     // TODO: recreating the tree every time messes with the expanded
     // state of ManagedTree, because it depends on item instances
     // being the same. The result is that if a source is added at a
     // later time, all expanded state is lost.
-    let sourceTree = this.state.sourceTree;
-    if (newSet.size > 0) {
-      for (const source of newSet) {
```

Comment:
i think this was done for performance reasons because recreating the tree every time is expensive.

Reply:
Tried the behavior of the tree in various scenarios and it is recreating anyway - so really not quite sure if there are any cases where the code brings performance benefits that have not been covered by the checks going before it.